packages/integration/lib/src/widgets/lists/feed/feed_list.dart

```diff
@@ -0,0 +1,672 @@
+          ),
+          child: LMFeedImage(
+            imageFile: media.mediaFile!,
+            style: const LMFeedPostImageStyle(
+              boxFit: BoxFit.contain,
+            ),
+          ),
+        );
+      } else if (media.mediaType == LMMediaType.document) {
+        return LMFeedTheme
+                .instance.theme.mediaStyle.documentStyle.documentIcon ??
+            LMFeedIcon(
+              type: LMFeedIconType.icon,
+              icon: Icons.picture_as_pdf,
+              style: LMFeedIconStyle(
+                color: Colors.red,
+                size: 35,
+                boxPadding: 0,
+              ),
+            );
+      } else {
+        return const SizedBox.shrink();
+      }
+    } else {
+      return const SizedBox.shrink();
+    }
+  }
+
+  @override
+  void initState() {
+    super.initState();
+    Bloc.observer = LMFeedBlocObserver();
+    userPostingRights = LMFeedUserUtils.checkPostCreationRights();
+    _addPaginationListener();
+  }
+
+  @override
+  void didUpdateWidget(covariant LMFeedList oldWidget) {
+    super.didUpdateWidget(oldWidget);
+    if (!listEquals(widget.selectedTopicIds, oldWidget.selectedTopicIds)) {
+      widget.pagingController.refresh();
+    }
+  }
+
+  void _addPaginationListener() {
+    widget.pagingController.addPageRequestListener(
+      (pageKey) {
+        _feedBloc.add(
+          LMFeedGetUniversalFeedEvent(
+            pageKey: pageKey,
+            topicsIds: widget.selectedTopicIds,
+          ),
+        );
+      },
+    );
+  }
+
+  void refresh() => widget.pagingController.refresh();
+
+  void updatePagingControllers(LMFeedState? state) {
+    if (state is LMFeedUniversalFeedLoadedState) {
+      List<LMPostViewData> listOfPosts = state.posts;
+
+      _feedBloc.users.addAll(state.users);
+      _feedBloc.topics.addAll(state.topics);
+      _feedBloc.widgets.addAll(state.widgets);
+
+      if (state.posts.length < 10) {
+        widget.pagingController.appendLastPage(listOfPosts);
+      } else {
+        widget.pagingController.appendPage(listOfPosts, state.pageKey + 1);
+      }
+    }
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return BlocListener<LMFeedBloc, LMFeedState>(
+      bloc: _feedBloc,
+      listener: (context, LMFeedState state) => updatePagingControllers(state),
+      child: ValueListenableBuilder(
+        valueListenable: rebuildPostWidget,
+        builder: (context, _, __) {
+          return PagedSliverList<int, LMPostViewData>(
+            pagingController: widget.pagingController,
+            builderDelegate: PagedChildBuilderDelegate<LMPostViewData>(
+              itemBuilder: (context, item, index) {
+                return LMFeedCore.widgetUtility.postWidgetBuilder(
+                    context, defPostWidget(context, feedThemeData, item), item);
+              },
+            ),
+          );
+        },
+      ),
+    );
+  }
+
+  LMFeedPostWidget defPostWidget(BuildContext context,
```

Comment:
We can use the widget from widgetUtility right?

Reply:
We are using widgets from widget utility, some widgets need to access state variable that is why we have def widgets in the screen itself